frame/staking/src/tests.rs

```diff
@@ -5128,6 +5131,9 @@ fn change_of_max_nominations() {
 			assert!(Nominators::<Test>::get(60).is_some());
 			assert_eq!(Staking::electing_voters(None).unwrap().len(), 3 + 1);
 
+			// TODO: This has to be fixed, we can't re-nominate without reporting previous
```

Comment:
@kianenigma I'm not sure what's the best course of action here, as discussed before we no longer want to support decreasing of max_nominations without a proper migration or any other vector bounds for that matter.
In this case here we have to restrict re-nomination if the previous nominations are not decodable. 
Otherwise the only solution is a hacky unbounded decode, which is not something we were aiming at.